utils/vscode/syntaxes/carbon.tmLanguage.json

```
@@ -0,0 +1,178 @@
+{
+	"$schema": "https://raw.githubusercontent.com/martinring/tmlanguage/master/tmlanguage.json",
+	"name": "carbon",
+	"patterns": [
+		{
+			"include": "#comments"
+		},
+		{
+			"include": "#operators"
+		},
+		{
+			"include": "#strings"
+		},
+		{
+			"include": "#numbers"
+		},
+		{
+			"include": "#reserved-words"
+		},
+		{
+			"include": "#operator-dedicated-keywords-statements"
+		},
+		{
+			"include": "#ctrl-statements"
+		},
+		{
+			"include": "#special-keywords"
+		},
+		{
+			"include": "#true-false"
+		},
+		{
+			"include": "#functions"
+		},
+		{
+			"include": "#customs"
+		}
+	],
+	"repository": {
+		"operators": {
+			"patterns": [
+				{
+					"name": "keyword.operator.new.carbon",
+					"match": "\\b(\\+|-|\\*|/|!)\\b"
+				}
+			]
+		},
+		"comments": {
+			"patterns": [
+				{
+					"name": "comment.line.double-dash.carbon",
+					"begin": "//",
+					"end": "\n"
+				}
+			]
+		},
+		"strings": {
+			"patterns": [
+				{
+					"name": "string.quoted.single.carbon",
+					"begin": "'",
+					"end": "'"
+				},
+				{
+					"name": "string.quoted.double.carbon",
+					"begin": "\"",
+					"end": "\""
+				},
+				{
+					"name": "string.quoted.triple.carbon",
+					"begin": "\"\"\"",
+					"end": "\"\"\""
+				}
+			]
+		},
+		"special-keywords": {
+			"patterns": [
+				{
+					"name": "keyword.control.less.carbon",
+					"match": "\\b(break|case|continue|default|else|if|for|match|return|returned|then|while)\\b"
+				}
+			]
+		},
+		"operator-dedicated-keywords-statements": {
+			"patterns": [
+				{
+					"name": "keyword.carbon",
+					"match": "\\b(abstract|adapt|addr|alias|and|api|as|auto|base|choice|class|constraint|destructor|extend|final|fn|forall|friend|impl|impls|import|in|interface|let|library|like|namespace|not|observe|or|override|package|partial|private|protected|require|Self|template|type|var|virtual|where|_)\\b"
+				}
+			]
+		},
+		"reserved-words": {
+			"patterns": [
+				{
+					"name": "support.type.carbon",
+					"match": "\\b(As|bool|Carbon\\.Int|Carbon\\.UInt|f16|f32|f64|f128|i8|i16|i32|i64|i128|i256|Slice|String|StringView|type|u8|u16|u32|u64|u128|u256)\\b"
+				}
+			]
+		},
+		"true-false": {
+			"patterns": [
+				{
+					"name": "keyword.operator.new.carbon",
```

```suggestion
					"name": "keyword.operator.new.carbon",
```
The name ".operator.new" is a surprise to me here, for bool literals.